tests/net/sntp/Kconfig

```diff
@@ -0,0 +1,15 @@
+# Copyright (c) 2024 Embeint Inc
+
+config SNTP_SERVER_ADDRESS
+	string "SNTP server address"
+	default "pool.ntp.org"
+
+config SNTP_SERVER_PORT
+	int "SNTP server port"
+	default 123
+
+config SNTP_SERVER_TIMEOUT_MS
+	int "Duration to wait for response"
+	default 2000
```

Comment:
Typically these tests are self contained and do not connect to outside world. One exception is
`tests/net/lib/lwm2m/interop` which connects to outside host but I think it can only be run manually.

One option is to move the tests to `samples/net` where the programs are run and can connect to outside world. In that case the these vars could be prefixed by `NET_SAMPLE_` or similar in order to emphasize that these are only for tests/samples and not for the SNTP library.

If we want to keep the tests under `tests/net`, we could have hard coded values from server or simulate the returns from server.

Reply:
+1 for that.

Reply:
So the choice is between not testing the behavior in CI (samples/net) and not testing against an actual server (tests/net)?
While the second is less bad IMO, I have no idea how to setup the test environment to receive the request and respond with a hardcoded response, and no real desire to write a simulated SNTP server either.

I was surprised when I saw no tests for SNTP in tree, I think I understand why now :)

Reply:
> I was surprised when I saw no tests for SNTP in tree, I think I understand why now :)

Yes, typically if there is desire to test against a server, then creating a test to `tests/net` is quite difficult. And in those cases testing can be done via a sample.
There are Docker based testing created in samples/net (search for docker-test.sh files). See https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/net/README.txt for details.

Reply:
There is also this issue https://github.com/zephyrproject-rtos/zephyr/issues/29764 to track Docker network tests.